webview-ui/src/Kaito/KaitoModels.tsx

```diff
@@ -0,0 +1,156 @@
+import { useState } from "react";
+import styles from "./KaitoModels.module.css";
+import kaitoSupporterModel from "../../../resources/kaitollmconfig/kaitollmconfig.json";
+import { VSCodeDivider, VSCodeLink } from "@vscode/webview-ui-toolkit/react";
+import { vscode2 } from "./state";
+
+function capitalizeFirstLetter(text: string) {
+    return text.charAt(0).toUpperCase() + text.slice(1);
+}
+
+function extractModelSource(url: string) {
+    const match = url.match(/https:\/\/huggingface\.co\/([^/]+)/);
+    return match ? match[1] : "Unknown";
+}
+
+export function KaitoModels() {
+    const modelFamilies: { [key: string]: string[] } = kaitoSupporterModel.modelsupported;
+    const modelDetails = kaitoSupporterModel.modelDetails;
+
+    // Identify the first family key
+    const firstFamilyKey = Object.keys(modelFamilies)[0];
+
+    // Set the initial state of openDropdowns to have the first family key set to true
+    const [openDropdowns, setOpenDropdowns] = useState<{ [key: string]: boolean }>({
+        [firstFamilyKey]: true,
+    });
+    const [selectedModel, setSelectedModel] = useState<string | null>(null);
+
+    const toggleDropdown = (family: string) => {
+        setOpenDropdowns((prev) => ({
+            ...prev,
+            [family]: !prev[family],
+        }));
+    };
+
+    const getModelDetails = (modelName: string) => {
+        return modelDetails.find((model) => model.modelName === modelName);
+    };
+
+    const handleModelClick = (model: string) => {
+        setSelectedModel(model);
+    };
+
+    const selectedModelDetails = getModelDetails(selectedModel !== null ? selectedModel : "");
+
+    function stringOrUndefined(s: string | undefined): string {
+        return s !== undefined ? s : "";
+    }
+
+    function generateCRD() {
+        const name = stringOrUndefined(selectedModelDetails?.modelName);
+        const yaml = `apiVersion: kaito.sh/v1alpha1
+kind: Workspace
+metadata:
+  name: workspace-${name}
+resource:
+  instanceType: "${stringOrUndefined(selectedModelDetails?.minimumGpu)}"
+  labelSelector:
+    matchLabels:
+      apps: ${name}
+inference:
+  preset:
+    name: "${name}"`;
+        vscode2.postGenerateCRDRequest({ model: yaml });
+        return;
+    }
+
+    return (
+        <div className={styles.mainDiv}>
+            <h2>Create a KAITO Workspace</h2>
```

Comment:
👍 Important note: Lets add clusterName here, if user has many, its hard to distinguish between screen, also add same cluster name for the createWorkspace as well. Thanks heaps!

Reply:
I agree I was looking for which cluster did I enabled Kaito lol.. :)